# 
# Submitted:
#
# Kai-Uwe Bloem, 2004-04-05
#
# Error: 
#
# Incorrect mmap return code
#
# Description: 
#
# <pointer to discussions on the net>
#
# State:
# 
# <upstream state of this problem>
#

--- ppp-2.4.1/pppd/tdb.c	Tue Apr  4 08:27:13 2000
+++ /home/kub/src/ppp-2.4.1/pppd/tdb.c	Mon Apr  5 11:42:00 2004
@@ -210,6 +210,7 @@
 	tdb->map_ptr = (void *)mmap(NULL, tdb->map_size, 
 				    tdb->read_only?PROT_READ:PROT_READ|PROT_WRITE,
 				    MAP_SHARED | MAP_FILE, tdb->fd, 0);
+        if (tdb->map_ptr == (void *)-1) tdb->map_ptr = NULL;
 #endif	
 	return 0;
 }
@@ -373,6 +374,7 @@
             tdb->map_ptr = (void *)mmap(NULL, tdb->map_size, 
                                         PROT_READ|PROT_WRITE,
                                         MAP_SHARED | MAP_FILE, tdb->fd, 0);
+            if (tdb->map_ptr == (void *)-1) tdb->map_ptr = NULL;
         }
 #endif
 
@@ -1180,6 +1182,7 @@
             tdb.map_ptr = (void *)mmap(NULL, st.st_size, 
                                        tdb.read_only? PROT_READ : PROT_READ|PROT_WRITE,
                                        MAP_SHARED | MAP_FILE, tdb.fd, 0);
+            if (tdb.map_ptr == (void *)-1) tdb.map_ptr = NULL;
         }
 #endif
 
